datapusher/jobs.py

```diff
@@ -426,14 +426,13 @@ def push_to_datastore(task_id, input, dry_run=False):
         logger.info('Done.')
         ct = 'application/csv'
 
-    read_exception = None
     try:
         table_set = messytables.any_tableset(tmp, mimetype=ct, extension=ct)
     except messytables.ReadError as e:
         read_exception = e
 
     # try again with format inferred from url
-    if not table_set.tables:
+    if not (hasattr(table_set, 'tables') and table_set.tables):
```

Comment:
That's better done with:
`if not table_set or not table_set.tables`
or statments are lazy evaluated so when `table_set` is None the latter part `not table_set.tables` won't be evaluated preventing NameError.